src/Fuse/ShaderNode.cs

```diff
@@ -43,8 +45,8 @@ private void BuildSource(StringBuilder theSourceBuilder, HashSet<int> theHashes)
                 if (!(child is AbstractShaderNode input)) return;
                
                 ((AbstractShaderNode)child).BuildSource(theSourceBuilder, theHashes);
-            });
-            if (!SourceCode.IsNullOrEmpty() && theHashes.Add(GetHashCode()))
+            });
+            if (!Stride.Core.Extensions.EnumerableExtensions.IsNullOrEmpty(SourceCode))
```

Comment:
this could be `string.IsNullOrWhitespace()`

Reply:
and `&& theHashes.Add(GetHashCode())` is missing?

Reply:
upsi...
those 2 are solved by the latest commits